framework/test/unit/modules/nft/method.spec.ts

```diff
@@ -101,4 +136,150 @@ describe('NFTMethods', () => {
 			await expect(method.getLockingModule(methodContext, nftID)).resolves.toEqual(lockingModule);
 		});
 	});
+
+	describe('destroy', () => {
+		let existingNFT: { nftID: any; owner: any };
+		let lockedExistingNFT: { nftID: any; owner: any };
+		let escrowedNFT: { nftID: any; owner: any };
+
+		beforeEach(async () => {
+			existingNFT = {
+				owner: utils.getRandomBytes(LENGTH_ADDRESS),
+				nftID: utils.getRandomBytes(LENGTH_NFT_ID),
+			};
+
+			lockedExistingNFT = {
+				owner: utils.getRandomBytes(LENGTH_ADDRESS),
+				nftID: utils.getRandomBytes(LENGTH_NFT_ID),
+			};
+
+			escrowedNFT = {
+				owner: utils.getRandomBytes(LENGTH_CHAIN_ID),
+				nftID: utils.getRandomBytes(LENGTH_NFT_ID),
+			};
+
+			await nftStore.save(methodContext, existingNFT.nftID, {
+				owner: existingNFT.owner,
+				attributesArray: [],
+			});
+
+			await userStore.set(methodContext, userStore.getKey(existingNFT.owner, existingNFT.nftID), {
+				lockingModule: NFT_NOT_LOCKED,
+			});
+
+			await nftStore.save(methodContext, lockedExistingNFT.nftID, {
+				owner: lockedExistingNFT.owner,
+				attributesArray: [],
+			});
+
+			await userStore.set(
+				methodContext,
+				userStore.getKey(lockedExistingNFT.owner, lockedExistingNFT.nftID),
+				{
+					lockingModule: 'token',
+				},
+			);
+
+			await nftStore.save(methodContext, escrowedNFT.nftID, {
+				owner: escrowedNFT.owner,
+				attributesArray: [],
+			});
+
+			await userStore.set(methodContext, userStore.getKey(escrowedNFT.owner, escrowedNFT.nftID), {
+				lockingModule: NFT_NOT_LOCKED,
+			});
+		});
+
+		it('should fail and emit Destroy event if NFT does not exist', async () => {
+			const address = utils.getRandomBytes(LENGTH_ADDRESS);
+
+			await expect(method.destroy(methodContext, address, nftID)).rejects.toThrow(
+				'NFT substore entry does not exist',
+			);
+
+			checkEventResult<DestroyEventData>(
+				methodContext.eventQueue,
+				1,
+				DestroyEvent,
+				0,
+				{
+					address,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+		});
+
+		it('should fail and emit Destroy event if NFT is not owned by the provided address', async () => {
+			const notOwner = utils.getRandomBytes(LENGTH_ADDRESS);
+
+			await expect(method.destroy(methodContext, notOwner, existingNFT.nftID)).rejects.toThrow(
+				'Not initiated by the NFT owner',
+			);
+
+			checkEventResult<DestroyEventData>(
+				methodContext.eventQueue,
+				1,
+				DestroyEvent,
+				0,
+				{
+					address: notOwner,
+					nftID: existingNFT.nftID,
+				},
+				NftEventResult.RESULT_INITIATED_BY_NONOWNER,
+			);
+		});
+
+		it('should fail and emit Destroy event if NFT is escrowed', async () => {
+			await expect(
+				method.destroy(methodContext, escrowedNFT.owner, escrowedNFT.nftID),
+			).rejects.toThrow();
```

Comment:
```suggestion
			await expect(
				method.destroy(methodContext, escrowedNFT.owner, escrowedNFT.nftID),
			).rejects.toThrow('NFT is escrowed to another chain');
```
maybe also include the error message